### PR TITLE
Validate `bbs2gh` env vars in the generated script

### PR DESCRIPTION
**Pull request by dylan-smith**
- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created)
- [x] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

Implements part of #629 

Added some powershell to the start of the script generated by `bbs2gh generate-script` that validates the proper environment variables are set before executing the rest of the script.

This one was a bit trickier than `ado2gh` because depending on what args you pass to `generate-script` the script will expect different sets of env vars. This was the pseudocode I came up with before coding this up:
```
always validate GH_PAT
always validate BBS_PASSWORD unless args.Kerberos
if args.BbsUsername.IsNullOrWhitespace() and !args.Kerberos validate BBS_USERNAME
if args.AwsBucketName.HasValue() or args.AwsRegion.HasValue()
	Validate AWS_ACCESS_KEY_ID
	Validate AWS_SECRET_ACCESS_KEY
if AwsBucketName.IsNull and AwsRegion.IsNull
	Validate AZURE_STORAGE_CONNECTION_STRING
if SmbUser.HasValue
	Validate SMB_PASSWORD
```

These are all the various validation messages:
```
if (-not $env:GH_PAT) {
    Write-Error "GH_PAT environment variable must be set to a valid GitHub Personal Access Token with the appropriate scopes. For more information see https://docs.github.com/en/migrations/using-github-enterprise-importer/preparing-to-migrate-with-github-enterprise-importer/managing-access-for-github-enterprise-importer#creating-a-personal-access-token-for-github-enterprise-importer"
    exit 1
} else {
    Write-Host "GH_PAT environment variable is set and will be used to authenticate to GitHub."
}

if (-not $env:BBS_USERNAME) {
    Write-Error "BBS_USERNAME environment variable must be set to a valid user that will be used to call BBS API's to generate a migration archive."
    exit 1
} else {
    Write-Host "BBS_USERNAME environment variable is set and will be used to authenticate to BBS APIs."
}

if (-not $env:BBS_PASSWORD) {
    Write-Error "BBS_PASSWORD environment variable must be set to a valid password that will be used to call BBS API's to generate a migration archive."
    exit 1
} else {
    Write-Host "BBS_PASSWORD environment variable is set and will be used to authenticate to BBS APIs."
}

if (-not $env:AZURE_STORAGE_CONNECTION_STRING) {
    Write-Error "AZURE_STORAGE_CONNECTION_STRING environment variable must be set to a valid Azure Storage Connection String that will be used to upload the migration archive to Azure Blob Storage."
    exit 1
} else {
    Write-Host "AZURE_STORAGE_CONNECTION_STRING environment variable is set and will be used to upload the migration archive to Azure Blob Storage."
}

if (-not $env:AWS_ACCESS_KEY_ID) {
    Write-Error "AWS_ACCESS_KEY_ID environment variable must be set to a valid AWS Access Key ID that will be used to upload the migration archive to AWS S3."
    exit 1
} else {
    Write-Host "AWS_ACCESS_KEY_ID environment variable is set and will be used to upload the migration archive to AWS S3."
}

if (-not $env:AWS_SECRET_ACCESS_KEY) {
    Write-Error "AWS_SECRET_ACCESS_KEY environment variable must be set to a valid AWS Secret Access Key that will be used to upload the migration archive to AWS S3."
    exit 1
} else {
    Write-Host "AWS_SECRET_ACCESS_KEY environment variable is set and will be used to upload the migration archive to AWS S3."
}

if (-not $env:SMB_PASSWORD) {
    Write-Error "SMB_PASSWORD environment variable must be set to a valid password that will be used to download the migration archive from your BBS server using SMB."
    exit 1
} else {
    Write-Host "SMB_PASSWORD environment variable is set and will be used to download the migration archive from your BBS server using SMB."
}
```

Note: I updated this PR to target another PR branch (instead of main) so the diff is correct for reviewers. Will retarget it to main once that other PR merges.

### CODE REVIEW

File: src/bbs2gh/Handlers/GenerateScriptCommandHandler.cs [hunk 1]
```diff
@@ -232,5 +255,54 @@ private void ValidateOptions(GenerateScriptCommandArgs args)
...
+    private const string VALIDATE_GH_PAT = @"
+if (-not $env:GH_PAT) {
+    Write-Error ""GH_PAT environment variable must be set to a valid GitHub Personal Access Token with the appropriate scopes. For more information see https://docs.github.com/en/migrations/using-github-enterprise-importer/preparing-to-migrate-with-github-enterprise-importer/managing-access-for-github-enterprise-importer#creating-a-personal-access-token-for-github-enterprise-importer""
+    exit 1
+} else {
+    Write-Host ""GH_PAT environment variable is set and will be used to authenticate to GitHub.""
+}";
+    private const string VALIDATE_BBS_USERNAME = @"
+if (-not $env:BBS_USERNAME) {
+    Write-Error ""BBS_USERNAME environment variable must be set to a valid user that will be used to call BBS API's to generate a migration archive.""
```
Comment (timrogers) [inline src/bbs2gh/Handlers/GenerateScriptCommandHandler.cs:268]:
```suggestion
    Write-Error ""BBS_USERNAME environment variable must be set to a valid user that will be used to call Bitbucket Server/Data Center APIs to generate a migration archive.""
```

File: src/bbs2gh/Handlers/GenerateScriptCommandHandler.cs [hunk 2]
```diff
@@ -232,5 +255,54 @@ private void ValidateOptions(GenerateScriptCommandArgs args)
...
+    exit 1
+} else {
+    Write-Host ""BBS_USERNAME environment variable is set and will be used to authenticate to BBS APIs.""
```
Comment (timrogers) [inline src/bbs2gh/Handlers/GenerateScriptCommandHandler.cs:271]:
```suggestion
    Write-Host ""BBS_USERNAME environment variable is set and will be used to authenticate to Bitbucket Server/Data Center APIs.""
```

File: src/bbs2gh/Handlers/GenerateScriptCommandHandler.cs (hunk 1 above)
Comment (timrogers) [reply src/bbs2gh/Handlers/GenerateScriptCommandHandler.cs:268]: I'll leave you to apply this change elsewhere in the file.

File: src/bbs2gh/Handlers/GenerateScriptCommandHandler.cs (hunk 2 above)
Comment (timrogers) [reply src/bbs2gh/Handlers/GenerateScriptCommandHandler.cs:271]: I'll leave you to apply this change elsewhere in the file.

File: src/OctoshiftCLI.Tests/bbs2gh/Handlers/GenerateScriptCommandHandlerTests.cs [hunk 3]
```diff
@@ -82,7 +82,211 @@ public async Task No_Projects()
         await _handler.Handle(args);
 
         // Assert
-        _mockFileSystemProvider.Verify(m => m.WriteAllTextAsync(It.IsAny<string>(), It.Is<string>(script => TrimNonExecutableLines(script, 9, 0) == "")));
+        _mockFileSystemProvider.Verify(m => m.WriteAllTextAsync(It.IsAny<string>(), It.Is<string>(script => TrimNonExecutableLines(script, 33, 0) == "")));
+    }
+
+    [Fact]
+    public async Task Validates_Env_Vars()
```
Comment (timrogers) [inline src/OctoshiftCLI.Tests/bbs2gh/Handlers/GenerateScriptCommandHandlerTests.cs:89]: Do you think that we should have an integration test that runs the PowerShell script as well?

**Review by timrogers — CHANGES_REQUESTED** [review]
Thanks for picking this up! ❤️ I've suggested some tweaks to the error messages.

File: src/OctoshiftCLI.Tests/bbs2gh/Handlers/GenerateScriptCommandHandlerTests.cs (hunk 3 above)
Comment (dylan-smith) [reply src/OctoshiftCLI.Tests/bbs2gh/Handlers/GenerateScriptCommandHandlerTests.cs:89]: All our integration tests run the powershell script

**Review by ArinGhazarian — APPROVED** [review]

**Pull request merged**